### PR TITLE
Service instance sharing  tests

### PR DESCRIPTION
**Pull request by FelisiaM**
[#174254053](https://www.pivotaltracker.com/story/show/174254053)

### CODE REVIEW

File: spec/request/service_instances_spec.rb [hunk 1]
```diff
@@ -3360,57 +3185,511 @@ def create_bindings(instance, space:, count:)
...
+        include(
+          {
+            'detail' => 'Service instance not found',
+            'title' => 'CF-ResourceNotFound'
+          })
+      )
+    end
+
+    describe 'invalid request body' do
+      context 'when request is not a valid relationship' do
```
Comment (kirederik) [inline spec/request/service_instances_spec.rb:3286]: maybe "when the request body is invalid"?

File: spec/request/service_instances_spec.rb [hunk 2]
```diff
@@ -3360,57 +3185,511 @@ def create_bindings(instance, space:, count:)
...
+          {
+            'data' => [
+              { 'guid' => no_access_target_space.guid },
+              { 'guid' => target_space_1.guid }
+            ]
+          }
+        end
+
+        it 'responds with 422' do
+          api_call.call(space_dev_headers)
```
Comment (kirederik) [inline spec/request/service_instances_spec.rb:3369]: one thing it's not clear from this test is whether the service instance was shared or not with the space the user has access to. Should we add another `it` to cover that (either `it 'does not share with any of the spaces'` or `it 'shares with the other spaces'` or something like that)

File: spec/request/service_instances_spec.rb [hunk 3]
```diff
@@ -3495,51 +3774,6 @@ def enable_sharing!
     let!(:service_instance2) { VCAP::CloudController::ManagedServiceInstance.make(space: space, name: 'redis') }
     let!(:service_instance3) { VCAP::CloudController::ManagedServiceInstance.make(space: another_space, name: 'mysql') }
 
-    describe 'POST /v3/service_instances/:guid/relationships/shared_spaces' do
```
Comment (kirederik) [inline spec/request/service_instances_spec.rb:3498]: I guess we want to delete the entire `unrefactored` describe that this one is in?

Comment (FelisiaM) [reply spec/request/service_instances_spec.rb:3498]: still have delete tests to finish  so will remove unrefactored and the controller unit test after that

File: spec/request/service_instances_spec.rb (hunk 2 above)
Comment (FelisiaM) [reply spec/request/service_instances_spec.rb:3370]: added an expectation to check that it was not shared at all

File: spec/request/service_instances_spec.rb (hunk 3 above)
Comment (FelisiaM) [reply spec/request/service_instances_spec.rb:3498]: I have removed all POST related tests

**Pull request merged**